internal/contentcache/contentcache.go

```diff
@@ -51,6 +52,38 @@ func (c *ContentCache) NewTempFile(rc io.ReadCloser) (gcsx.TempFile, error) {
 	return gcsx.NewTempFile(rc, c.tempDir, c.mtimeClock)
 }
 
+// Add or replace creates a new cache file or updates an existing cache file
+func (c *ContentCache) AddOrReplace(cacheObjectKey *CacheObjectKey, generation int64, rc io.ReadCloser) (gcsx.TempFile, error) {
+	if cacheObject, exists := c.fileMap[*cacheObjectKey]; exists {
+		cacheObject.Destroy()
+	}
+	metadata := &gcsx.TempFileObjectMetadata{
+		BucketName: cacheObjectKey.BucketName,
+		ObjectName: cacheObjectKey.ObjectName,
+		Generation: generation,
+	}
+	file, err := c.NewCacheFile(rc, metadata)
+	if err != nil {
+		return nil, err
+	}
+	c.fileMap[*cacheObjectKey] = file
+	return file, err
+}
+
+// Get retrieves a file from the cache given the GCS object name and bucket name
+func (c *ContentCache) Get(cacheObjectKey *CacheObjectKey) (gcsx.TempFile, bool) {
+	file, exists := c.fileMap[*cacheObjectKey]
+	return file, exists
+}
+
+// Remove removes and destroys the specfied cache file and metadata on disk
+func (c *ContentCache) Remove(cacheObjectKey *CacheObjectKey) {
+	if _, exists := c.fileMap[*cacheObjectKey]; exists {
+		c.fileMap[*cacheObjectKey].Destroy()
```

Comment:
This is mentioned in a previous comment still unresolved. Unnecessary map lookup.